config/config-api/src/com/thoughtworks/go/config/EnvironmentVariableConfig.java

```diff
@@ -172,6 +172,10 @@ public void validateName(Map<String, EnvironmentVariableConfig> variableConfigMa
         String currentVariableName = name.toLowerCase();
         String parentDisplayName = validationContext.getParentDisplayName();
         CaseInsensitiveString parentName = getParentNameFrom(validationContext);
+        if(currentVariableName.startsWith(" ") || currentVariableName.endsWith(" ")){
```

Comment:
Damn, I thought all feedback was worked on, when i was pinged. My bad.
@GaneshSPatil can you use trim instead of this?